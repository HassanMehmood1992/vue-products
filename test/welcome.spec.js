import { mount, createLocalVue } from "@vue/test-utils";
import Vuetify from "vuetify";
import Vuex from "vuex";

import Welcome from "@/components/welcome";

const localVue = createLocalVue();
localVue.use(Vuex);

let wrapper;

beforeEach(() => {
  let vuetify = new Vuetify();

  wrapper = mount(Welcome, {
    store: new Vuex.Store({
      state: { products: [] }
    }),
    localVue,
    vuetify,
    stubs: ["router-link"]
  });
});

afterEach(() => {
  wrapper.destroy();
});

describe("Welcome component", () => {
  test("is fully functional", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  test("should have welcome title", () => {
    const title = wrapper.find(".py-3.welcome.primary--text");

    expect(title.element.innerHTML).toBe("Welcome To Shophub");
  });

  test("should render sub title", () => {
    const title = wrapper.find(".text-h6");
    expect(title.element.innerHTML).toBe(
      "Enjoy our favorite products ! Its Awesome"
    );
  });
});
