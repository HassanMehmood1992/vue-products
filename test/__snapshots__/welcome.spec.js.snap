// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome component is fully functional 1`] = `
<div
  class="flex mx-auto sm6"
>
  <div
    class="mx-2 v-card v-card--flat v-sheet theme--light"
  >
    <div
      class="v-card__text pa-0"
    >
      <div
        class="flex text-center"
      >
        <img
          class="ma-3"
          src="undefinedgift.png"
          width="100px"
        />
         
        <div
          class="py-3 welcome primary--text"
        >
          Welcome To Shophub
        </div>
         
        <div
          class="text-h6"
        >
          Enjoy our favorite products ! Its Awesome
        </div>
      </div>
    </div>
  </div>
</div>
`;
